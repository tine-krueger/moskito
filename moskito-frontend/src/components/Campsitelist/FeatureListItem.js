import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import { nameReturn }from '../../services/featureService'

FeatureListItem.propTypes = {
    feature: PropTypes.object.isRequired
}

export default function FeatureListItem({feature}) {

    return <FeatureItem>{nameReturn(feature)}</FeatureItem>

}

const FeatureItem = styled.li `
    border-radius: 10px;
    display: inline-block;
    box-shadow: inset 4px 4px 6px 0 rgba(0, 0, 0, 0.2), 
                inset -3px -4px 6px 0 rgba(255, 255, 255, 0.3);
    padding: .5em 1em;
    margin: .5em;
`
