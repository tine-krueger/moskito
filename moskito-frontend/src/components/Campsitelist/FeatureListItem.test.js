import { render } from '@testing-library/react'
import 'jest-styled-components'
import FeatureListItem from './FeatureListItem'

const feature = {
    type: 'wlan'
}
describe('Feature List Item', () => {
    it('renders correctly', () => {
        const { container } = render(
            <FeatureListItem
                feature={feature}
            />
        )
        expect(container).toMatchSnapshot()
    })

    it('shows the correct feature name', () => {
        const { getByText } = render(
            <FeatureListItem
                feature={feature}
            />
        )
        expect(getByText('kein W-Lan')).toBeInTheDocument()
    })
})
