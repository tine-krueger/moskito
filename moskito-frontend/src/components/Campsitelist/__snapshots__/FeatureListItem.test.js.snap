// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature List Item renders correctly 1`] = `
.c0 {
  border-radius: 10px;
  display: inline-block;
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3);
  padding: .5em 1em;
  margin: .3em;
}

<div>
  <li
    class="c0"
  >
    kein W-Lan
  </li>
</div>
`;
